copy/copy.go

```diff
@@ -699,24 +699,61 @@ func checkImageDestinationForCurrentRuntime(ctx context.Context, sys *types.Syst
 		if err != nil {
 			return errors.Wrapf(err, "Error parsing image configuration")
 		}
-
-		wantedOS := runtime.GOOS
-		if sys != nil && sys.OSChoice != "" {
-			wantedOS = sys.OSChoice
+		platforms, err := platform.WantedPlatforms(sys)
+		if err != nil {
+			return errors.Wrapf(err, "error getting current platform information %#v", sys)
 		}
-		if wantedOS != c.OS {
-			logrus.Infof("Image operating system mismatch: image uses %q, expecting %q", c.OS, wantedOS)
+		warning := runtimeMatchWarning(c, platforms)
+		if warning != "" {
+			logrus.Info(warning)
 		}
+	}
+	return nil
+}
 
-		wantedArch := runtime.GOARCH
-		if sys != nil && sys.ArchitectureChoice != "" {
-			wantedArch = sys.ArchitectureChoice
+// runtimeMatchWarning returns an error message warning about the mismatch
+// between an image config and set of platforms by a runtime, or "" if the two match.
+func runtimeMatchWarning(cfg *imgspecv1.Image, platforms []imgspecv1.Platform) string {
```

Comment:
Nit: Using an error (instead of a string) as return value seems more idiomatic to me. Can we change it?